### ISSUE TITLE
couldn't find oab-utils - running in legacy mode, please report this to the developer

### ISSUE THREAD

**Issue by vi** [jensstein/oandbackup#222]
Meizu 15 Plus, Android 7.0 (Flyme 7.0.3.1A)

**Comment by vi** [comment]
Yes, it's version `0.3.2-x86 [suggested] [installed]` on aarch64 device.

**Comment by jensstein** [comment]
that seems like a bug on fdroid's side if they suggest an x86 version for your arm device.
at the same time there is a bug in the build configuration which affects arm64 so even if you get the right version from fdroid it's not going to work until the next version (0.3.3).
(also, you need arm64, not arm)
https://gitlab.com/fdroid/fdroiddata/issues/1366

**Comment by meijkl** [comment]
Just upgraded to 0.3.3 via F-Droid, still the same . Message "couldn't find oab-utilities - running in legacy mode ..."

My setup: Nexus 5x, LOS 15.1 with microg, no Gapps etc.

**Comment by jensstein** [comment]
@vi just verifying: are you running 0.3.2-arm, not 0.3.3-arm64? 0.3.3 is the latest release and your device needs the arm64 or the universal version.
but looking at the fdroid page there seems to be a problem with which versions they offer because the latest are
0.3.3-x86_64
0.3.2-x86
0.3.3-arm64
i'll contact them to find out if it's a problem i or they or we both have to fix.

if you can't get it from fdroid, there are packages on my gitlab repo: https://gitlab.com/jensstein/oandbackup/-/jobs/116043498/artifacts/browse 

@meijkl which abi version of the apk are you running? your nexus is 64bit as far as i know so you also need either 0.3.3-arm64 or 0.3.3-universal

**Comment by meijkl** [comment]
Sorry, I didn't see the x86 problem.

My case is special - I installed oandbackup via Nanodroid and the x86 version was automatically chosen (don't know why). Now I've reinstalled Nanodroid without oandbackup and installed it manually afterwards via F-Droid. But I had to choose the arm64 version manually, F-Droid automatically selected the x86-64 version.

My problem is solved - thanks for your efforts!

**Comment by DamienCassou** [comment]
I have a One+ 1 (bacon) and just installed 0.3.3-arm64 manually. I get 2 errors:

- `error copying asset` (same as https://gitlab.com/fdroid/fdroiddata/issues/1366#note_112428585), and
- couldn't find oab-utils

**Comment by DamienCassou** [comment]
My mistake: my architecture is arm, not arm64. I have no more problems, thank you.

**Comment by Kelvino9** [comment]
@roizcorp pick the arm64 version manually in F-Droid like all of us with arm64 Phones. F-Droid has problems detecting the correct architecture for oandbackup.

**Comment by roizcorp** [comment]
yep that works...thanks

**Comment by jensstein** [comment]
there is an issue on fdroid's gitlab repo where i mention this problem: https://gitlab.com/fdroid/fdroiddata/issues/1366

**Comment by jensstein** [comment]
the packagers at fdroid are reporting that the universal apk flavour doesn't work. it works for me but i can only test on armv7 devices so could i get some of you to try this apk and report if it works or not and which abi (arm, arm64, x86, or x86_64) your device has?
https://gitlab.com/jensstein/oandbackup/-/jobs/116206298/artifacts/file/apks/oandbackup-universal-debug.apk

and it seems like they rely on the output of the ```aapt``` command to determine which abi the built apk targets - this causes some difficulty since rust isn't supported in an splits.abi block in the gradle build configuration (as far as i can tell) and therefore the ```aapt``` output doesn't contain a ```native-code``` field. i can probably make a work-around but suggestions are welcome :)

**Comment by zzzardoz** [comment]
@jensstein: Your universal apk works without any issues here (Oneplus One/arm)!

**Comment by meijkl** [comment]
Just tried to install on my Nexus 6p/AICP Nougat/arm64. Failed with message "package seems to be corrupt".

On a Nexus 5x with LOS Pie prerelease the installation fails too with message "App not installed".

**Comment by Kelvino9** [comment]
Debug apk works fine on Redmi Note 4/ AICP Oreo/ arm64

@meijkl try to re-download with a different browser.

**Comment by meijkl** [comment]
Had to uninstall oandbackup, afterwards the installation worked.

**Comment by gitgt** [comment]
Same as Issue 212

**Comment by gitgt** [comment]
Please close this issue  if resolved.

**Comment by vi** [comment]
After manually picking arm64 version of oandbackup it works.

Shall F-Droid behaviour of picking the wrong version or misusing version string as architecture variant specified be in scope of this issue?

**Comment by tristan-k** [comment]
Is this fixed?

**Comment by vi** [comment]
Maybe the error message should be extended with a hint that user may be using the wrong version of oandbackup?

**Comment by chilliger** [comment]
I am still getting this message when opening the app. When I try to update the app through F-Droid I get this message:
![]()
<img src="https://user-images.githubusercontent.com/2310896/57571949-c7b1d880-7414-11e9-977f-b4f160bacc4c.png" width="250"> <img src="https://user-images.githubusercontent.com/2310896/57571974-2bd49c80-7415-11e9-809e-afb52f1504d8.png" width="250">
F-Droid thinks that I want to downgrade. What are my options here other than reinstalling and losing all configurations?

**Comment by DamienCassou** [comment]
I have the same issue.

**Comment by mvglasow** [comment]
@chilliger same here. 0.3.3-x86_64 has internal version code 4303 while 0.3.5-universal has 305, therefore F-Droid thinks you are downgrading when in fact you are upgrading. Not sure where the error originated, though.

Downgrade via F-Droid did not work for me as the Android package manager rejected the replacement package.

You could try downloading the 0.3.5-universal APK to a PC from F-Droid, then installing it via adb:

`adb install -r -d oandbackup.apk`

This requires ADB installed on your PC and enabled on your device. The command line options tell ADB to replace the existing package and force a downgrade.
